### PR TITLE
Add auto fix functionality

### LINKED ISSUE #10: Add --fix support

### PR DESCRIPTION
**Pull request by pranaygp**
Closes #10 

This is a feature we need for a project that we're soon launching to ensure that all our code has the appropriate copyright header line.

### CODE REVIEW

File: lib/rules/header.js [hunk 1]
```diff
@@ -25,6 +25,41 @@ function getLeadingComments(context, node) {
...
+function genCommentsRange(context, comments) {
+    var start = comments[0].range[0];
+    var end = comments.slice(-1)[0].range[1];
+    if (context.getSourceCode().text[end] === "\n") {
+        end++;
+    }
+    return [start, end];
+}
+
+function genPrependFixer(commentType, context, node, headerLines) {
```
Comment (Stuk) [inline lib/rules/header.js]: `context` isn't used here. Could you remove it?

File: lib/rules/header.js [hunk 2]
```diff
@@ -38,23 +73,32 @@ module.exports = function(context) {
     // If commentType is line then we want an array of the lines,
     // but if block then we want just a string
     var header, headerLines;
+    var shouldFix = false;
```
Comment (Stuk) [inline lib/rules/header.js]: Would this be better named `canFix`?

Either way, could you add a brief comment explaining that this is false if any lines are regexes?

File: lib/rules/header.js [hunk 3]
```diff
@@ -64,24 +108,56 @@ module.exports = function(context) {
...
             } else {
                 if (commentType === "line") {
                     if (leadingComments.length < headerLines.length) {
-                        context.report(node, "incorrect header");
+                        context.report({
+                            loc: node.loc,
+                            message: "incorrect header",
+                            data: {
+                                commentType: commentType
+                            },
```
Comment (Stuk) [inline lib/rules/header.js]: This data isn't used in this message

File: lib/rules/header.js [hunk 4]
```diff
@@ -64,24 +108,56 @@ module.exports = function(context) {
...
                     }
                     for (var i = 0; i < headerLines.length; i++) {
                         if (!match(leadingComments[i].value, headerLines[i])) {
-                            context.report(node, "incorrect header");
+                            context.report({
+                                loc: node.loc,
+                                message: "incorrect header",
+                                data: {
+                                    commentType: commentType
+                                },
```
Comment (Stuk) [inline lib/rules/header.js]: This data isn't used in this message

File: lib/rules/header.js [hunk 5]
```diff
@@ -64,24 +108,56 @@ module.exports = function(context) {
...
                     }
                 } else {
                     if (!match(leadingComments[0].value, header)) {
-                        context.report(node, "incorrect header");
+                        context.report({
+                            loc: node.loc,
+                            message: "incorrect header",
+                            data: {
+                                commentType: commentType
+                            },
```
Comment (Stuk) [inline lib/rules/header.js]: This data isn't used in this message

**Review by Stuk — CHANGES_REQUESTED** [review]
Thank you so much for the contribution! This is great!

I have a few comments in review.

`npm test` and `npm run lint` seem to be failing. Could you look at those and make the needed changes?

Also, I appreciate that yarn is fast and cool but I'm not using it on this project, so could you remove the yarn.lock file?

**Comment by pranaygp** [comment]
Gotcha!

I'm a little surprised `npm test` didn't pass. seems to work locally. Could you give that a shot again? 

Some of the lint issues seems to not be from my code. I assume it was always broken. Would you still like me to push a fix within this PR to existing lint issues? 

Also, I'm adding `yarn.lock` to the .gitignore to make that explicit

**Comment by pranaygp** [comment]
I opened up #13 to address lint issues since they weren't from the code changes in this PR. It's branched off this to ensure easy merging after this PR is merged :D

**Review by Stuk — APPROVED** [review]

**Comment by Stuk** [comment]
Thank you for the code! Something was wrong with my local Node install, tests are all passing now (confirmed by Travis!)

**Comment by Stuk** [comment]
Published as v1.2.0

**Pull request merged**